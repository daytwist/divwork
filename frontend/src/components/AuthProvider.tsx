import Cookies from "js-cookie";
import {
  createContext,
  Dispatch,
  FC,
  ReactNode,
  SetStateAction,
  useContext,
  useMemo,
  useState,
} from "react";

type Props = {
  children: ReactNode;
};

type AuthContextValue = {
  isSignedIn: boolean;
  setIsSignedIn: Dispatch<SetStateAction<boolean>>;
};

export const AuthContext = createContext<AuthContextValue>(
  {} as AuthContextValue
);

export const AuthProvider: FC<Props> = ({ children }) => {
  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);

  const AuthProviderValue = useMemo(
    () => ({ isSignedIn, setIsSignedIn }),
    [isSignedIn, setIsSignedIn]
  );

  return (
    <AuthContext.Provider value={AuthProviderValue}>
      {children}
    </AuthContext.Provider>
  );
};

export const GetIsSignedIn = () => {
  const { isSignedIn, setIsSignedIn } = useContext(AuthContext);

  if (
    Cookies.get("_access_token") &&
    Cookies.get("_client") &&
    Cookies.get("_uid")
  ) {
    setIsSignedIn(true);
    console.log(isSignedIn);
  } else {
    setIsSignedIn(false);
    console.log(isSignedIn);
  }
};
