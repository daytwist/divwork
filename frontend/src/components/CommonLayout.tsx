import { css, Global } from "@emotion/react";
import { Box, Container, Grid } from "@mui/material";
import { FC } from "react";
import { RouteProps } from "react-router-dom";
import Header from "./Header";

const CommonLayout: FC<RouteProps> = ({ children }) => {
  const global = css`
    * {
      margin: 0;
    }
  `;

  return (
    <div>
      <header>
        <Header />
      </header>
      <main>
        <Global styles={global} />
        <Box m={2} pt={4}>
          <Container maxWidth="lg">
            <Grid container justifyContent="center">
              <Grid item>{children}</Grid>
            </Grid>
          </Container>
        </Box>
      </main>
    </div>
  );
};

export default CommonLayout;
