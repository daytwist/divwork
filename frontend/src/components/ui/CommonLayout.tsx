import { FC } from "react";
import { RouteProps } from "react-router-dom";
import { css, Global } from "@emotion/react";
import {
  Box,
  Container,
  createTheme,
  responsiveFontSizes,
  ThemeProvider,
  Toolbar,
} from "@mui/material";
import Grid2 from "@mui/material/Unstable_Grid2/Grid2";
import { jaJP } from "@mui/x-data-grid";
import MenuBars from "./MenuBars";
import { AlertSnackbar } from "./AlertSnackbar";
import { Footer } from "./Footer";

const CommonLayout: FC<RouteProps> = ({ children }) => {
  let theme = createTheme(
    {
      palette: {
        primary: {
          main: "#1de9b6",
          light: "#6effe8",
          dark: "#00b686",
        },
        secondary: {
          main: "#5a6abf",
          light: "#8d98f2",
          dark: "#23408e",
        },
      },
      breakpoints: {
        values: {
          xs: 0,
          sm: 600,
          md: 768,
          lg: 1025,
          xl: 1536,
        },
      },
    },
    jaJP
  );
  theme = responsiveFontSizes(theme);

  const styles = {
    "*": css`
      margin: 0;
    `,

    main: css`
      display: flex;
      flex-direction: column;
      min-height: calc(103vh - 68.5px - 50px);
    `,

    footer: css`
      margin-top: auto;
    `,
  };

  return (
    <div>
      <ThemeProvider theme={theme}>
        <Global styles={styles} />
        <AlertSnackbar />
        <Box sx={{ display: "flex" }}>
          <MenuBars />
          <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
            <Toolbar />
            <Container sx={{ py: 3 }}>
              <Grid2 container justifyContent="center">
                <Grid2>{children}</Grid2>
              </Grid2>
            </Container>
          </Box>
        </Box>
        <Footer />
      </ThemeProvider>
    </div>
  );
};

export default CommonLayout;
