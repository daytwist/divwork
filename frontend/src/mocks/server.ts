// eslint-disable-next-line import/no-extraneous-dependencies
import { setupWorker } from "msw";
import { handlers } from "./handlers";

export const server = setupWorker(...handlers);
