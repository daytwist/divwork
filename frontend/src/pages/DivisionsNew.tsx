import { ChangeEvent, FC, useEffect, useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import Cookies from "js-cookie";
import { Button, Container, TextField } from "@mui/material";
import { AxiosRequestConfig, AxiosResponse } from "axios";
import { axiosInstance } from "../utils/axios";
import { divisionTask, TasksResponse, DivisionsCreateResponse } from "../types";

const DivisionsNew: FC = () => {
  const params = useParams<{ id: string }>();
  const navigate = useNavigate();

  const [task, setTask] = useState<divisionTask>({
    title: "",
    description: "",
    deadline: "",
    priority: "",
    is_done: false,
    user_id: 0,
    parent_id: 0,
  });

  const [comment, setComment] = useState<string>("");

  const handleInputChange = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    const { name, value } = event.target;
    setTask({ ...task, [name]: value });
  };

  const handleDivisionsCreate = () => {
    const options: AxiosRequestConfig = {
      url: `/tasks/${params.id}/divisions`,
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "access-token": Cookies.get("_access_token") || "",
        client: Cookies.get("_client") || "",
        uid: Cookies.get("_uid") || "",
      },
      data: { task, division: { comment } },
    };

    axiosInstance(options)
      .then((res: AxiosResponse<DivisionsCreateResponse>) => {
        console.log(res);

        if (res.status === 200) {
          navigate(`/users/${res.data.division.user_id}`, { replace: false });
        }
      })
      .catch((err) => console.log(err));
  };

  const options: AxiosRequestConfig = {
    url: `/tasks/${params.id}/divisions/new`,
    method: "GET",
    headers: {
      "access-token": Cookies.get("_access_token") || "",
      client: Cookies.get("_client") || "",
      uid: Cookies.get("_uid") || "",
    },
  };

  useEffect(() => {
    axiosInstance(options)
      .then((res: AxiosResponse<TasksResponse>) => {
        console.log(res.data);
        setTask(res.data.task);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  return (
    <Container maxWidth="sm">
      <h1>タスクを分担する</h1>
      <div>
        <TextField
          label="タイトル"
          variant="standard"
          name="title"
          value={task.title}
          onChange={handleInputChange}
        />
      </div>
      <br />
      <div>
        <TextField
          label="詳細"
          variant="standard"
          multiline
          rows={3}
          name="description"
          value={task.description}
          onChange={handleInputChange}
        />
      </div>
      <br />
      <div>
        <label>
          納期
          <input
            type="text"
            name="deadline"
            value={task.deadline}
            onChange={handleInputChange}
          />
        </label>
      </div>
      <br />
      <div>
        <label>
          優先度
          <input
            name="priority"
            value={task.priority}
            onChange={handleInputChange}
          />
        </label>
      </div>
      <br />
      <div>
        <label>
          送信先ユーザー
          <input name="user_id" onChange={handleInputChange} />
        </label>
      </div>
      <br />
      <div>
        <TextField
          label="送信コメント"
          variant="standard"
          value={comment}
          onChange={(event) => setComment(event.target.value)}
        />
      </div>
      <br />
      <Button variant="contained" type="submit" onClick={handleDivisionsCreate}>
        送信
      </Button>
    </Container>
  );
};

export default DivisionsNew;
