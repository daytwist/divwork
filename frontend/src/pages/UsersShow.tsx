import { useContext, FC, useEffect, useState, SyntheticEvent } from "react";
import { Link } from "react-router-dom";
import { Box, Button, Grid, Tab, Tabs, Typography } from "@mui/material";
import { AuthContext } from "../providers/AuthProvider";
import { useFetchUser } from "../hooks/useFetchUser";
import { Task } from "../types";
import { TasksTable } from "../components/TasksTable";

const UsersShow: FC = () => {
  const { user, unfinishedTasks, finishedTasks } = useFetchUser();
  const { currentUser } = useContext(AuthContext);

  const [tasks, setTasks] = useState<Task[]>([]);
  const [isFinished, setIsFinished] = useState<boolean>(false);

  const [value, setValue] = useState("unfinished");

  const handleSwitchTasks = (event: SyntheticEvent, newValue: string) => {
    setValue(newValue);

    if (newValue === "finished") {
      setIsFinished(true);
      setTasks(finishedTasks);
      console.log(isFinished);
    } else {
      setIsFinished(false);
      setTasks(unfinishedTasks);
      console.log(isFinished);
    }
  };

  useEffect(() => {
    if (isFinished) {
      setTasks(finishedTasks);
    } else {
      setTasks(unfinishedTasks);
    }
  }, [unfinishedTasks, finishedTasks]);

  return (
    <div>
      <Grid container direction="column" spacing={3}>
        <Grid item>
          <Typography variant="h4" component="div" data-testid="users-show-h4">
            {user?.name ? `${user.name}のタスク` : ""}
          </Typography>
        </Grid>
        <Grid item>
          <Grid
            container
            direction="row"
            justifyContent="space-between"
            alignItems="center"
          >
            {user?.id === currentUser?.id ? (
              <Grid item>
                <Button
                  variant="contained"
                  type="button"
                  component={Link}
                  to="/tasks/new"
                >
                  新規作成
                </Button>
              </Grid>
            ) : (
              <br />
            )}
            <Grid item>
              <Box sx={{ width: "100%" }}>
                <Tabs
                  value={value}
                  onChange={handleSwitchTasks}
                  aria-label="tabs"
                >
                  <Tab value="unfinished" label="未了" />
                  <Tab value="finished" label="完了済み" />
                </Tabs>
              </Box>
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <TasksTable
            user={user}
            tasks={tasks}
            setTasks={setTasks}
            isFinished={isFinished}
          />
        </Grid>
      </Grid>
    </div>
  );
};

export default UsersShow;
