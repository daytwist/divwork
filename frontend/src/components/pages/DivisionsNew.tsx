import { ChangeEvent, useEffect, useState } from "react";
import {
  Button,
  Divider,
  FormControl,
  InputLabel,
  MenuItem,
  OutlinedInput,
  Select,
  Stack,
  TextField,
  Typography,
} from "@mui/material";
import Grid2 from "@mui/material/Unstable_Grid2/Grid2";
import AutoAwesomeIcon from "@mui/icons-material/AutoAwesome";
import { User } from "../../types/userTypes";
import { DivisionTask } from "../../types/taskTypes";
import { TasksForm } from "../models/task/TasksForm";
import { BackIconButton } from "../ui/BackIconButton";
import { usePostDivision } from "../../hooks/division/usePostDivision";
import { useFetchNewDivision } from "../../hooks/division/useFetchNewDivision";
import { LoadingColorRing } from "../ui/LoadingColorRing";

export const DivisionsNew = () => {
  const [divisionData, isLoading] = useFetchNewDivision();

  const [task, setTask] = useState<DivisionTask>({
    title: "",
    description: "",
    priority: "",
    rate_of_progress: 0,
    parent_id: 0,
  });

  const [deadline, setDeadline] = useState<Date | null>(new Date());
  const [comment, setComment] = useState<string>("");
  const [teamMembers, setTeamMembers] = useState<User[]>([]);
  const [teamMemberValue, setTeamMemberValue] = useState<string>("");

  const handleCreateDivision = usePostDivision({
    task,
    deadline,
    comment,
    teamMemberValue,
  });

  const ITEM_HEIGHT = 48;
  const ITEM_PADDING_TOP = 8;
  const MenuProps = {
    PaperProps: {
      style: {
        maxHeight: ITEM_HEIGHT * 3 + ITEM_PADDING_TOP,
        width: 150,
      },
    },
  };

  const handleInputChange = (
    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    const { name, value } = event.target;
    setTask({ ...task, [name]: value });
  };

  const handleDeadlineChange = (newValue: Date | null) => {
    setDeadline(newValue);
  };

  useEffect(() => {
    if (divisionData) {
      setTask(divisionData?.task);
      setDeadline(divisionData?.task.deadline);
      setTeamMembers(divisionData?.team_members);
    }
  }, [divisionData]);

  if (isLoading) {
    return <LoadingColorRing />;
  }

  return (
    <Grid2
      container
      direction="column"
      rowSpacing={3}
      sx={{ width: { xs: 300, sm: 500, lg: 700 } }}
    >
      <Grid2 xs={12}>
        <Stack direction="row" spacing={1} alignItems="center">
          <BackIconButton />
          <Typography gutterBottom variant="h4" component="div">
            タスクを分担する
          </Typography>
        </Stack>
      </Grid2>
      <Grid2 xs={12}>
        <TasksForm
          action="divisionsNew"
          task={task}
          onChange={handleInputChange}
          deadline={deadline}
          onChangeDeadline={handleDeadlineChange}
        />
      </Grid2>
      <Divider sx={{ my: { xs: 2, sm: 4 } }} />
      <Grid2 xs={12}>
        <FormControl required sx={{ width: 200 }} color="secondary">
          <InputLabel id="users_label">分担先ユーザー</InputLabel>
          <Select
            labelId="users_label"
            name="user_id"
            value={teamMemberValue}
            onChange={(event) => setTeamMemberValue(event.target.value)}
            input={<OutlinedInput label="分担先ユーザー" />}
            MenuProps={MenuProps}
          >
            {teamMembers.map((member) => (
              <MenuItem key={member.id} value={member.id.toString()}>
                {member.name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </Grid2>
      <Grid2 xs={12}>
        <TextField
          inputProps={{ maxLength: 100 }}
          label="分担コメント"
          variant="outlined"
          color="secondary"
          sx={{ width: "100%" }}
          helperText="100文字以内"
          value={comment}
          onChange={(event) => setComment(event.target.value)}
        />
      </Grid2>
      <Grid2 xs={12}>
        <Button
          data-testid="send-button"
          variant="contained"
          type="submit"
          onClick={handleCreateDivision}
          startIcon={<AutoAwesomeIcon />}
        >
          分担する
        </Button>
      </Grid2>
    </Grid2>
  );
};
