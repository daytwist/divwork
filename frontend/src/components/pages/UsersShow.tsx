import {
  useContext,
  useState,
  SyntheticEvent,
  useEffect,
  useCallback,
} from "react";
import { useParams } from "react-router-dom";
import { Box, Stack, Tab, Tabs } from "@mui/material";
import Grid2 from "@mui/material/Unstable_Grid2/Grid2";
import { GridRowId } from "@mui/x-data-grid";
import { TasksActionButtons } from "../models/user/TasksActionButtons";
import { AuthContext } from "../../providers/AuthProvider";
import { useUser } from "../../hooks/useUser";
import { TasksDataGrid } from "../models/user/TasksDataGrid";
import { DivisionsDataGrid } from "../models/user/DivisionsDataGrid";
import { LoadingColorRing } from "../ui/LoadingColorRing";
import { TabPanel } from "../ui/TabPanel";
import { UserNameHeader } from "../models/user/UserNameHeader";
import { BackIconButton } from "../ui/BackIconButton";

export const UsersShow = () => {
  const { currentUser } = useContext(AuthContext);
  const { id: userId } = useParams();
  const [reloadFlag, setReloadFlag] = useState(false);
  const [isFinished, setIsFinished] = useState(false);
  const [tabValue, setTabValue] = useState(0);
  const [selectionModel, setSelectionModel] = useState<GridRowId[]>([]);
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const [userData, isLoading, error] = useUser(userId, reloadFlag);

  const tabProps = (index: number) => {
    return {
      id: `tab-${index}`,
      "aria-controls": `tabpanel-${index}`,
    };
  };

  const handleSwitchTasks = useCallback(
    (event: SyntheticEvent, newValue: number) => {
      setTabValue(newValue);

      if (newValue === 0) {
        setIsFinished(false);
        setSelectionModel([]);
      } else if (newValue === 1) {
        setIsFinished(true);
        setSelectionModel([]);
      }
    },
    []
  );

  const unfinishedTasksDataGrid = (
    <TasksDataGrid
      isFinished={false}
      user={userData?.user}
      tasks={userData?.unfinished_tasks}
      selectionModel={selectionModel}
      setSelectionModel={setSelectionModel}
    />
  );

  const finishedTasksDataGrid = (
    <TasksDataGrid
      isFinished
      user={userData?.user}
      tasks={userData?.finished_tasks}
      selectionModel={selectionModel}
      setSelectionModel={setSelectionModel}
    />
  );

  const divisionsDataGrid = (
    <DivisionsDataGrid divisions={userData?.divisions} />
  );

  useEffect(() => {
    if (userData?.unfinished_tasks && tabValue === 0) {
      setIsFinished(false);
      setSelectionModel([]);
    } else if (userData?.finished_tasks && tabValue === 1) {
      setIsFinished(true);
      setSelectionModel([]);
    }
  }, [userData?.unfinished_tasks, userData?.finished_tasks]);

  useEffect(() => {
    if (userId) {
      setTabValue(0);
      setIsFinished(false);
    }
  }, [userId]);

  useEffect(() => {
    if (error) {
      console.log(error);
    }
  }, [error]);

  if (isLoading) {
    return <LoadingColorRing />;
  }

  return (
    <div>
      <Grid2
        container
        direction="column"
        rowSpacing={1}
        alignContent="center"
        alignItems="center"
        data-testid="users-show-page"
      >
        <Grid2 xs={12}>
          <Stack direction="row" spacing={1} alignItems="center">
            <BackIconButton />
            <UserNameHeader user={userData?.user} />
          </Stack>
        </Grid2>
        <Grid2 xs={12}>
          <Stack
            direction="row"
            justifyContent="space-between"
            alignItems="flex-end"
          >
            {userData?.user?.id === currentUser?.id ? (
              <TasksActionButtons
                selectionModel={selectionModel}
                isFinished={isFinished}
                flag={reloadFlag}
                setFlag={setReloadFlag}
                tabValue={tabValue}
              />
            ) : (
              <br />
            )}
            <Box sx={{ borderBottom: 1, borderColor: "divider" }}>
              <Tabs
                value={tabValue}
                onChange={handleSwitchTasks}
                textColor="inherit"
              >
                <Tab label="未了" {...tabProps(0)} />
                <Tab label="完了済み" {...tabProps(1)} />
                <Tab label="分担履歴" {...tabProps(2)} />
              </Tabs>
            </Box>
          </Stack>
        </Grid2>
        <Grid2
          xs={12}
          sx={{ width: { xs: 330, sm: 560, md: 850, lg: 900, xl: 1050 } }}
        >
          <TabPanel value={tabValue} index={0}>
            {unfinishedTasksDataGrid}
          </TabPanel>
          <TabPanel value={tabValue} index={1}>
            {finishedTasksDataGrid}
          </TabPanel>
          <TabPanel value={tabValue} index={2}>
            {divisionsDataGrid}
          </TabPanel>
        </Grid2>
      </Grid2>
    </div>
  );
};
