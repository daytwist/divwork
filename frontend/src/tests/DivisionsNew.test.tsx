/* eslint-disable @typescript-eslint/await-thenable */
/* eslint-disable testing-library/no-unnecessary-act */
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { rest } from "msw";
import { act } from "react-dom/test-utils";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { server } from "../mocks/server";
import { AuthProvider } from "../providers/AuthProvider";
import DivisionsNew from "../components/pages/DivisionsNew";
import UsersShow from "../components/pages/UsersShow";
import { SnackbarProvider } from "../providers/SnackbarProvider";
import CommonLayout from "../components/ui/CommonLayout";

describe("DivisionsNew", () => {
  test("分担作成ページ", async () => {
    server.use(
      rest.get("/auth/sessions", (req, res, ctx) => {
        return res(
          ctx.status(200),
          ctx.json({
            is_signed_in: true,
            current_user: {
              email: "test@example.com",
              uid: "test@example.com",
              id: 1,
              provider: "email",
              allow_password_change: false,
              name: "USER_1",
              nickname: null,
              image: null,
              team_id: 1,
            },
          })
        );
      })
    );

    render(
      <MemoryRouter initialEntries={["/tasks/1/divisions/new"]}>
        <AuthProvider>
          <SnackbarProvider>
            <CommonLayout>
              <Routes>
                <Route path="/users/:id" element={<UsersShow />} />
                <Route
                  path="/tasks/:id/divisions/new"
                  element={<DivisionsNew />}
                />
              </Routes>
            </CommonLayout>
          </SnackbarProvider>
        </AuthProvider>
      </MemoryRouter>
    );

    // タスクが表示される
    expect(await screen.findByDisplayValue("TASK_3")).toBeInTheDocument();
    await waitFor(() => {
      expect(screen.queryByText("USER_2")).not.toBeInTheDocument();
    });

    act(() => {
      userEvent.click(screen.getByLabelText("分担先ユーザー"));
    });

    // チームメンバーが表示される
    expect(await screen.findByText("USER_2")).toBeInTheDocument();

    await act(() => {
      userEvent.click(screen.getByText("USER_2"));
      userEvent.type(screen.getByLabelText("分担コメント"), "Thank you");
      userEvent.click(screen.getByTestId("send-button"));
    });

    // UsersShowページに遷移する
    expect(await screen.findByTestId("users-show-h4")).toBeInTheDocument();
    expect(
      await screen.findByText("分担タスクを送信しました")
    ).toBeInTheDocument();
  }, 8000);
});
