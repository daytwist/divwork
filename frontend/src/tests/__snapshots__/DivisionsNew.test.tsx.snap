// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivisionsNew スナップショット 1`] = `
<div>
  <h1>
    Divisions#New
  </h1>
  <div>
    <label>
      タイトル
      <input
        name="title"
        onChange={[Function]}
        value=""
      />
    </label>
  </div>
  <br />
  <div>
    <label>
      詳細
      <textarea
        name="description"
        onChange={[Function]}
        value=""
      />
    </label>
  </div>
  <br />
  <div>
    <label>
      納期
      <input
        name="deadline"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <br />
  <div>
    <label>
      優先度
      <input
        name="priority"
        onChange={[Function]}
        value=""
      />
    </label>
  </div>
  <br />
  <div>
    <label>
      送信先ユーザー
      <input
        name="user_id"
        onChange={[Function]}
      />
    </label>
  </div>
  <br />
  <div>
    <label>
      コメント
      <input
        onChange={[Function]}
        value=""
      />
    </label>
  </div>
  <br />
  <button
    onClick={[Function]}
    type="submit"
  >
    分担する
  </button>
</div>
`;
